crates/proto/src/op/serial_message.rs

```diff
@@ -12,6 +12,7 @@ use crate::error::ProtoResult;
 use crate::op::Message;
 
 /// A DNS message in serialized form, with either the target address or source address
+#[derive(Clone)]
```

Comment:
Is this still necessary? It feels like we should be able to get by with `Arc`-wrapping?